workshops/reddit_viewer/README.md

```diff
@@ -0,0 +1,126 @@
+---
+
+![The end result](https://cloud-fz3kdxk80.vercel.app/image.png)
+
+Python combined with Flask is an amazing way to understand how to make a website backend. Understanding how to make a backend is very simple and in this tutorial, we will be making our own Reddit web client that fetches directly from the reddit API.
+
+# Step 1: Setup
+
+Since this is just Python there isn't much setup. Just go to [repl.it](https://repl.it) and make a new repl. Make sure you import the 2 libraries required for this, and 2 for keeping your api key safe.
+
+```py
+import praw
+from dotenv import load_dotenv
+import os
+from flask import Flask, render_template, request
+
+```
+
+In the same folder as your main.py, make a file called `.env`. This is where we will store the api keys. Also go back to the main.py file and add these four lines to load your env file and flask,
+
+```py
+load_dotenv()
+app = Flask(__name__)
+
+if __name__ == '__main__':
+  app.run(host='0.0.0.0' , port = 9060)
+```
+
+The first one just finds and loads the your .env file and the second one says that this python file is the main file to run flask in. The third line is just doing a simple if check that will then run flask on localhost with the port being 9060.
+
+# Step 2: Getting Reddit API Key
+
+Getting your api key is very simple, go to the [Reddit Apps page](https://www.reddit.com/prefs/apps/) and click on `create another app...` at the bottom. ![The create another app button at the bottom](https://cloud-i2087nmyc.vercel.app/image.png)
+
+You should have a little section with `name`, `type`, `description`, `about url`, `redirect uri`. We only need the `name`, `type`, and `redirect uri`. For the name fill it with any name you want. For the type choose `script` and for the redirect uri, this doesn't matter, so just put `http://www.example.com/unused/redirect/uri`. ![The finished application creation](https://cloud-f67e393ze.vercel.app/image.png)
+
+In your repl.it open your .env file and make a variable called `SECRET` and another one called `ID`. Now go back to your reddit apps section and copy the secret from the secret part to your SECRET variable in your env file. Also copy the section just under your personal use script to your ID variable.
+
+![](https://cloud-119djxpws.vercel.app/image.png)
+
+We have the authentication setup which is easier done than said. Now one last step.
+
+We need to make a reddit variable that connects to the reddit api. Make a reddit = praw.Reddit() function. Now the next steps all go into the parenthesis. All it requires is three things, `client_id` which is your `"ID"` variable. So, do `client_id=os.getenv("ID"),`. The next one is client_secret. `client_secret=os.getenv("SECRET"),`. The final one is user_agent which is just a fancy name for what browser on what device. We will use a sample one as it doesn't actually matter which one you use. `user_agent="Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0")`
```

Comment:
Unless you have a specific reason for formatting the paragraph this way and adding an image with the code below it, I would just add all of this code in a Python code block. Written in paragraph form, this is very confusing to me.

Reply:
Done!